main/src/ca/uwaterloo/flix/language/phase/unification/set/SetFormula.scala

```diff
@@ -749,10 +768,10 @@ object SetFormula {
       var currentInsts = insts ++
         setElemOneOpt(elemPos, Univ) ++
         setElemOneOpt(elemNeg, Empty) ++
-        cstsPos.map(_.c -> Univ) ++
-        cstsNeg.map(_.c -> Empty) ++
-        varsPos.map(_.x -> Univ) ++
-        varsNeg.map(_.x -> Empty)
+        cstsPos.iterator.map(_.c -> Univ) ++
```

Comment:
cstsPos.map(_.c -> Unit) requires ordering on `(Int, SetFormula)` and `cstsPos.toSet.map(..)` reintroduces hash sets

Reply:
What about this comment?